src/app/Upload.svelte

```diff
@@ -41,11 +53,26 @@
     const input = event.target as HTMLInputElement;
     if (input !== null && input.files) {
       for (let file of input.files) {
-        files.set(file.name, file);
-        alert(file.name);
+        uploadFile(file);
       }
     }
   };
+
+  function uploadFile(file: File) {
+    files.set(file.name, file);
+    /* Useless assignment so that Svelte picks it up: */
+    // See: https://svelte.dev/tutorial/updating-arrays-and-objects
+    files = files;
+  }
+
+  /**
+   */
+  function workInProgressCreateModel() {
+    compileModel(files).then((sourceCode: string) => {
+      generatedCode = sourceCode;
```

Comment:
**[Question]**: I not sure if we are going to leverage on pure function programming, but mutating outside scope is definitely not a good practice, it harder to test and might be buggy

Reply:
Haha, so because of Svelte's reactive assignments, it needs to be done. I agree, mutation should be avoided whenever possible, but Svelte has _indirect control-flow` where [an assignment to a variable in the component's scope gets compiled to an `$$invalidate()` call](https://lihautan.com/compile-svelte-in-your-head-part-2/#invalidate), which updates the DOM.

Reply:
awesome, time for me to dive deeper into learning Svelte